File: src/main/java/pl/arturczopek/mycoach/model/add/NewPrice.java
package pl.arturczopek.mycoach.model.add;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.io.Serializable;
import java.sql.Date;

/**
 * @Author Artur Czopek
 * @Date 05-11-2016
 */

@Data
@AllArgsConstructor
public class NewPrice implements Serializable {

    private static final long serialVersionUID = -2267814709244502404L;

    @JsonProperty(required = true)
    private long productId;

    @JsonProperty(required = true)
    private float value;

    @JsonProperty(required = true)
    private float quantity;

    private String place;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date priceDate;
}

File: src/main/kotlin/pl/arturczopek/mycoach/model/add/NewPrice.kt
package pl.arturczopek.mycoach.model.add;

import com.fasterxml.jackson.annotation.JsonFormat
import java.sql.Date

/**
 * @Author Artur Czopek
 * @Date 05-11-2016
 */

data class NewPrice(
        val productId: Long,
        val value: Float,
        val quantity: Float,
        val place: String?,
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd") val priceDate: Date
)
